sys/include/tsrb.h

```diff
@@ -67,6 +67,17 @@ static inline void tsrb_init(tsrb_t *rb, uint8_t *buffer, unsigned bufsize)
     rb->writes = 0;
 }
 
+/**
+ * @brief        Reset a tsrb.
+ * @param[out]   rb        Ringbuffer to operate on
+ */
+static inline void tsrb_reset(tsrb_t *rb)
+{
+    unsigned irq_state = irq_disable();
+    tsrb_init(rb, rb->buf, rb->size);
```

Comment:
```suggestion
    rb->reads = rb->writes;
```

would also work. It might be beneficial to keep `rb->reads` and `rb->writes` monotonically growing (except for overflows) for debugging (e.g. to check the total number of bytes written to the tsrb). But I agree that it is more obvious that a call to `tsrb_init()` will clear the buffer when reading the code.

I think that a modern compiler will emit equally efficient code for both versions.

Reply:
Feel free to address or not @HendrikVE squash and trigger ci in either case!